nixos/modules/tasks/filesystems.nix

```diff
@@ -290,11 +299,10 @@ in
       "/dev" = { fsType = "devtmpfs"; options = [ "nosuid" "strictatime" "mode=755" "size=${config.boot.devSize}" ]; };
       "/dev/shm" = { fsType = "tmpfs"; options = [ "nosuid" "nodev" "strictatime" "mode=1777" "size=${config.boot.devShmSize}" ]; };
       "/dev/pts" = { fsType = "devpts"; options = [ "nosuid" "noexec" "mode=620" "gid=${toString config.ids.gids.tty}" ]; };
-    } // optionalAttrs (!config.boot.isContainer) {
       # systemd-nspawn populates /sys by itself, and remounting it causes all
       # kinds of weird issues (most noticeably, waiting for host disk device
       # nodes).
-      "/sys" = { fsType = "sysfs"; options = [ "nosuid" "noexec" "nodev" ]; };
+      "/sys" = { enabled = !config.boot.isContainer; fsType = "sysfs"; options = [ "nosuid" "noexec" "nodev" ]; };
     };
```

Comment:
Ah, yeah it should be `enable`, I'll change it.

And yes, we should move it to `container.nix`. Except there is no such file :) I started to try to find the right file to add it to, but got confused about the division between `nixos/modules/profiles` and `nixos/modules/virtualisation`. I'll see if I can prepare another PR for cleaning that up a bit, so I can add a proper `libvirt-lxc` config there too.

OK, I see if I can add it to `fileSystems` too.